File: 1570-final-prices-with-a-special-discount-in-a-shop/final-prices-with-a-special-discount-in-a-shop.js
/**
 * @param {number[]} prices
 * @return {number[]}
 */
var finalPrices = function(prices) {
    let res=[...prices];
    for(let i=0;i<prices.length;i++){
        res[i]=prices[i];
        for(let j=i+1;j<prices.length;j++){
            if(prices[i]>=prices[j]){
                res[i]=prices[i]-prices[j];
                break;
            }
        }
    }
    return res;
};

File: 1570-final-prices-with-a-special-discount-in-a-shop/final-prices-with-a-special-discount-in-a-shop.py
class Solution(object):
    def finalPrices(self, prices):
        """
        :type prices: List[int]
        :rtype: List[int]
        """
        res=[0]*len(prices)
        for i in range(len(prices)):
            res[i]=prices[i]
            for j in range(i+1,len(prices)):
                if prices[i]>=prices[j]:
                    res[i]=prices[i]-prices[j]
                    break
        return res